x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -309,6 +314,37 @@ func calculateRate(resultMetricsetFields common.MapStr, monitoringState string)
 	}
 }
 
+func addHostFields(resultMetricsetFields common.MapStr, rootFields common.MapStr, instanceID string) common.MapStr {
+	hostRootFields := common.MapStr{}
+	hostRootFields.Put("host.id", instanceID)
+
+	// If there is no instance name, use instance ID as the host.name
+	hostName, err := rootFields.GetValue("host.name")
+	if err == nil && hostName != nil {
+		hostRootFields.Put("host.name", hostName)
+	} else {
+		hostRootFields.Put("host.name", instanceID)
+	}
+
+	hostFieldTable := map[string]string{
+		"cpu.total.pct":       "host.cpu.pct",
+		"network.in.bytes":    "host.network.in.bytes",
+		"network.out.bytes":   "host.network.out.bytes",
+		"network.in.packets":  "host.network.in.packets",
+		"network.out.packets": "host.network.out.packets",
+		"diskio.read.bytes":   "host.disk.read.bytes",
+		"diskio.write.bytes":  "host.diskio.write.bytes",
```

Comment:
Use the same field name in both:
```suggestion
		"diskio.read.bytes":   "host.disk.read.bytes",
		"diskio.write.bytes":  "host.disk.write.bytes",
```